### PR TITLE
{ai}[foss/2022b] PyTorch v2.0.1

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

Requires (including rebuild!)
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19180
- #19253

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusml7 - Linux RHEL 7.6, POWER, 8335-GTX, 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/Flamefire/f0e7be8bf7b9299977dcfef226456953 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2
CORE_CNT=16

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-rockylinux-88 - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/SebastianAchilles/95ca63f9b91cc427f12629328044f7de for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 5 out of 6 (1 easyconfigs in total)
node3100.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/9130be6704b11e9034d9f739f8604aca for a full test report.

**Comment by Flamefire** [comment]
@boegel @branfosj I see a pattern there (visible thanks to the updated EasyBlock):
```
test_consistency_SparseBSC_sgn_cpu_uint8
test_consistency_SparseBSC_sign_cpu_uint8
test_consistency_SparseBSR_sgn_cpu_uint8
test_consistency_SparseBSR_sign_cpu_uint8
test_consistency_SparseCSC_sgn_cpu_uint8
test_consistency_SparseCSC_sign_cpu_uint8
test_consistency_SparseCSR_sgn_cpu_uint8
test_consistency_SparseCSR_sign_cpu_uint8
test_contig_vs_every_other_sgn_cpu_uint8
test_contig_vs_every_other_sign_cpu_uint8
test_non_contig_sgn_cpu_uint8
test_non_contig_sign_cpu_uint8
test_reference_numerics_normal_sgn_cpu_uint8
test_reference_numerics_normal_sign_cpu_uint8
test_reference_numerics_small_sgn_cpu_uint8
test_reference_numerics_small_sign_cpu_uint8
test_sparse_consistency_sgn_cpu_uint8
test_sparse_consistency_sign_cpu_uint8
```

Looks like an issue with the `sgn` and `sign` functions for uint8/bool tensors so very likely the same underlying issue/bug. Not sure why this happens only for those 2 test reports. Any ideas what they have in common? Maybe only ones with AVX512?

**Comment by branfosj** [comment]
The `inductor/test_torchinductor_opinfo` failures are identical to https://github.com/easybuilders/easybuild-easyconfigs/pull/19066#issuecomment-1780608273


The traceback is similar for all of them.
```
======================================================================
FAIL: test_sparse_consistency_sgn_cpu_uint8 (__main__.TestSparseUnaryUfuncsCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/dev/shm/branfosj/tmp-up-EL8/eb-o_7bs8k4/tmp3dgmc2zq/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 401, in instantiated_test
    result = test(self, **param_kwargs)
  File "/dev/shm/branfosj/tmp-up-EL8/eb-o_7bs8k4/tmp3dgmc2zq/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 851, in test_wrapper
    return test(*args, **kwargs)
  File "/dev/shm/branfosj/build-up-EL8/PyTorch/2.0.1/foss-2022b/pytorch-v2.0.1/test/test_sparse.py", line 3953, in test_sparse_consistency
    self.assertEqual(_sparse_to_dense(output), expected)
  File "/dev/shm/branfosj/tmp-up-EL8/eb-o_7bs8k4/tmp3dgmc2zq/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2979, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Tensor-likes are not equal!

Mismatched elements: 39 / 400 (9.8%)
Greatest absolute difference: 254 at index (17, 2)
Greatest relative difference: 0.9960784316062927 at index (17, 2)
```

I'm rebuilding now with `--skip-test-step` and I'll look to see if I can find anything more useful.

**Comment by Flamefire** [comment]
I'd be interested in the output for `test_non_contig_sgn_cpu_uint8` or `test_non_contig_sign_cpu_uint8`

My current analysis:
- `sgn/sign` is supposed to return one of `(-1,0,1)` depending on the sign of the input
- inputs of `uint8_t` can only be in range `[0,255]` hence the only valid outputs or `0` or `1`
- a difference of `254` in the output is likely an overflow, i.e. either the reference or the result had an intermediate value of `-1` where the other is  `1` (`uint8_t(-1) == 255 - 1` -> `uint8_t(-1) - 1 = 254`)

Not sure how that can happen though

**Comment by branfosj** [comment]
```
======================================================================
FAIL: test_non_contig_sgn_cpu_uint8 (__main__.TestUnaryUfuncsCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/PyTorch/2.0.1-foss-2022b/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 401, in instantiated_test
    result = test(self, **param_kwargs)
  File "/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/PyTorch/2.0.1-foss-2022b/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 851, in test_wrapper
    return test(*args, **kwargs)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-easyconfigs/easybuild/easyconfigs/tmp/pytorch-v2.0.1/test/test_unary_ufuncs.py", line 366, in test_non_contig
    self.assertEqual(op_contig, op_non_contig)
  File "/rds/projects/2017/branfosj-rse/easybuild/EL8-ice/software/PyTorch/2.0.1-foss-2022b/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2979, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Tensor-likes are not equal!

Mismatched elements: 952 / 1024 (93.0%)
Greatest absolute difference: 254 at index (0,)
Greatest relative difference: 254.0 at index (0,)
```

In `test_non_contig`, it fails when `shape = (1024,)`. I tested further and it fails with >= 64 and passes when the tensor is smaller than that.

`op(contig, **torch_kwargs)` and `op(non_contig, **torch_kwargs)` for `shape = (64,)` where `Mismatched elements: 61 / 64 (95.3%)`
```
tensor([255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,   0, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,   0, 255, 255, 255, 255, 255, 255,   0, 255, 255, 255, 255, 255, 255, 255, 255, 255], dtype=torch.uint8)
tensor([  1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0,   1,   1,   1,   1,   1,   1,   0,   1,   1,   1,   1,   1,   1,   1,   1,   1], dtype=torch.uint8)
```
So, the `0` are in the same place and are the matched elements. The mismatched ones are all `255` vs `1`.

**Comment by Flamefire** [comment]
> In test_non_contig, it fails when shape = (1024,). I tested further and it fails with >= 64 and passes when the tensor is smaller than that.

So your system is an AVX512 system? (64*8bit=512bit) so it has a bug in the AVX512 vectorized code.

I'll see if I can find that. Strange though, that it passes those tests for 2022a.

I opened draft PRs for 2.1.0 where I just ported the patches from here and removed those that are no longer required. Those are not tested yet, but maybe you can given them a try just to see if it compiles and succeeds on those tests at least.

**Comment by branfosj** [comment]
> So your system is an AVX512 system?

Yes

```
processor       : 71
vendor_id       : GenuineIntel
cpu family      : 6
model           : 106
model name      : Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz
stepping        : 6
microcode       : 0xd0003a5
cpu MHz         : 3500.000
cache size      : 55296 KB
physical id     : 1
siblings        : 36
core id         : 35
cpu cores       : 36
apicid          : 198
initial apicid  : 198
fpu             : yes
fpu_exception   : yes
cpuid level     : 27
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities
bugs            : spectre_v1 spectre_v2 spec_store_bypass swapgs mmio_stale_data
bogomips        : 4818.77
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 57 bits virtual
power management:
```

**Comment by Flamefire** [comment]
@branfosj or @boegel I'd like to report that to PyTorch as this is likely a bug there. Or in our 2022b toolchain (which I hope not). Does your GCCcore/12.2.0 installation include the `GCCcore-12.2.0_fix-vectorizer.patch`?

If so and you can spare the time I'd like you to try the following (on such an AVX512 node):

- switch to this build env: `eb --dump-env PyTorch-2.0.1-foss-2022b.eb` & `source PyTorch-2.0.1-foss-2022b.env`
- Create a virtualenv: `python -m venv --system-site-packages test-env & source test-env/bin/activate`
- install the official PyTorch module (`pip install torch==2.0.1`, later same but with `pip install torch==2.1.0`)
- (Shallow) clone pytorch `git clone --depth 1 --branch v2.0.1 https://github.com/pytorch/pytorch.git`
- Run the single failing test: `cd test && python test_unary_ufuncs.py -k test_non_contig_sign_cpu_uint8`

I hope this reproduces on the pip installed 2.0.1 (sanity check for us) and expect it to also fail on 2.1.0 (git checkout of the latter not required, only pip)

After this we can blame this on PyTorch and just skip the tests leaving the bug in the installed module as-if it was pip-installed if that is acceptable. At least I couldn't find anything wrong, so it might be a compiler bug given it works for 2022a

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusi8018 - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor, 8 x NVIDIA NVIDIA A100-SXM4-40GB, 470.57.02, Python 2.7.5
See https://gist.github.com/Flamefire/d8b71cd29d1b3a3fc47cfa9c481edd7a for a full test report.

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**FAILED**
Build succeeded for 7 out of 8 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/VRehnberg/f67612fbe10aafa8c2c639a72b926d28 for a full test report.

**Comment by VRehnberg** [comment]
@Flamefire 
> @branfosj or @boegel I'd like to report that to PyTorch as this is likely a bug there. Or in our 2022b toolchain (which I hope not). Does your GCCcore/12.2.0 installation include the `GCCcore-12.2.0_fix-vectorizer.patch`?

Yes.

>     * Run the single failing test: `cd test && python test_unary_ufuncs.py -k test_non_contig_sign_cpu_uint8`
>  
> I hope this reproduces on the pip installed 2.0.1 (sanity check for us) and expect it to also fail on 2.1.0 (git checkout of the latter not required, only pip)

For 2.0.1 (and 2.1.0 on SkyLake machine, didn't try on IceLake) I'm not seeing any test errors for this test after trying on two different machines with the following avx512 flags listed by lscpu (don't know which matter here or what all the differences mean):
 - IceLake machine:
```
avx512f
avx512dq
avx512ifma
avx512cd
avx512bw
avx512vl
avx512vbmi
avx512_vbmi2
avx512_vnni
avx512_bitalg
avx512_vpopcntdq
```
 - SkyLake machine:
``` 
avx512f
avx512dq
avx512cd
avx512bw
avx512vl
```

**Comment by Flamefire** [comment]
I reviewed the relevant code and used AVX512 intrinsics and haven't found anything wrong with them. Also a `python -c 'import torch; print(torch.__config__.show())'` for the official Python 2.0.1 package shows it is build with GCC 9.3.

Given that this EC and test works with foss/2022a I'd suspect this to be a bug in GCC 12 -.-

Short of disabling AVX512 by patching it out I don't know what to do...

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis-s1 - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Silver 4314 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/VRehnberg/c0dcdfe963e1a21326261e7569f1df1a for a full test report.

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/VRehnberg/adfca3a103ccbd73fd27d86b5c787b4a for a full test report.

**Comment by VRehnberg** [comment]
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis-s1 - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Silver 4314 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/VRehnberg/788e060921d3e1efc064878af92fdb08 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusml7 - Linux RHEL 7.6, POWER, 8335-GTX, 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/Flamefire/33b6f5810cfa0038962e2962f1d9e38f for a full test report.

**Comment by sassy-crick** [comment]
I am currently trying to do a testbuild but I noticed these modules are also dragged in:
``` 
GCCcore-10.2.0.eb
zlib-1.2.11-GCCcore-10.2.0.eb
help2man-1.47.16-GCCcore-10.2.0.eb
M4-1.4.18-GCCcore-10.2.0.eb
Bison-3.7.1-GCCcore-10.2.0.eb
flex-2.6.4-GCCcore-10.2.0.eb
binutils-2.35-GCCcore-10.2.0.eb
ncurses-6.2-GCCcore-10.2.0.eb
bzip2-1.0.8-GCCcore-10.2.0.eb
cURL-7.72.0-GCCcore-10.2.0.eb
XZ-5.2.5-GCCcore-10.2.0.eb
libarchive-3.4.3-GCCcore-10.2.0.eb
CMake-3.18.4-GCCcore-10.2.0.eb
```
Most of them are build-dependencies so ok I guess but clearly `bzip2` for example is not. Are we shooting us in our foot here?

**Comment by Flamefire** [comment]
> Most of them are build-dependencies so ok I guess but clearly `bzip2` for example is not. Are we shooting us in our foot here?

I don't understand why we would be shooting our feet by loading bzip2. Can you explain what issue you see with that?

FWIW: This dependency is pulled in by Python which requires it for the bz2 Package in the Python stdlib according to the comment in that EC. So it looks entirely reasonable to me.

**Comment by sassy-crick** [comment]
> > Most of them are build-dependencies so ok I guess but clearly `bzip2` for example is not. Are we shooting us in our foot here?
> 
> I don't understand why we would be shooting our feet by loading bzip2. Can you explain what issue you see with that?
> 
> FWIW: This dependency is pulled in by Python which requires it for the bz2 Package in the Python stdlib according to the comment in that EC. So it looks entirely reasonable to me.

Apologies, I should have been a bit clearer: As you mentioned Python, that is the module which is being loaded:
`Python/3.10.8-GCCcore-12.2.0`
and that is the module for `bzip`:
`bzip2-1.0.8-GCCcore-10.2.0.eb`
Different toolchains. That is what I was after. I only noticed it as when I done `--fetch` I had some issues with downloading one package on the cluster and thus I was watching what it does. I hope that makes sense now.

**Comment by Flamefire** [comment]
@sassy-crick 
Ah now I understand and it puzzled me too but I forgot about that particularity.  
TLDR: All is fine, just ignore it.

We depend on `Catch2` which is a header only library installed via CMake which is hence a build dependency. And that particular CMake we use as a build dependency for that (system-level) Catch2 EC is from GCCcore 10.2.0 hence the dependency on bzip2 from 10.2.0. As it is a build dependency it won't show up in the module dependencies. In fact even Catch2 is only a build dependency (of pybind11) so that won't show up either.

**Comment by sassy-crick** [comment]
I have uploaded a test report [here](https://gist.github.com/sassy-crick/5b98573d01435421a81149f8ccbb54a4). 
Unfortunately, it seems the log-file is too large (53M) to put up as a Gist on GitHub but happy to share it if somebody needs it. 
I also got access to AMD Zen2 Rome if that would help.

**Comment by Flamefire** [comment]
That report is missing what exactly is failing. Is it the same `sgn`/`sign` tests as reported earlier due to AVX512 or something different? FWIW: I'm trying to get this running on our new Sapphire Rapids which support AVX512 so that I can check what is going wrong. However I'm actually quite...disappointed by GCC 12 which we newly use in 2022b as that seemingly introduced quite some regressions. E.g. the destructor issue causing crashes in the JIT test I worked around with the earlier patch but also a bogus warning when compiling for the rapids I'm currently trying to just ignore/suppress. So it might well be that the AVX512 issue is also another regression in GCC 12.

**Comment by VRehnberg** [comment]
Fwiw, I haven't tried https://github.com/easybuilders/easybuild-easyconfigs/pull/19066 yet so sent out two test builds of those as well now, reports should come in that thread in a few hours.

**Comment by VRehnberg** [comment]
> Fwiw, I haven't tried #19066 yet so sent out two test builds of those as well now, reports should come in that thread in a few hours.

Success on both SkyLake and IceLake.

**Comment by Flamefire** [comment]
Found another bug in the compiler causing the AVX512 issues. Added another patch hoping that works...

**Comment by Flamefire** [comment]
@branfosj @boegel The compiler bug leading to the sign/sgn failures should be fixed by reinstalling with https://github.com/easybuilders/easybuild-easyconfigs/pull/19180 which should make this pass

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
n1519 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/f67744c6f9ee9febf6d9d333eb4d8afd for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0105u03b - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/8a762b78b462a396159926f67a94b345 for a full test report.

**Comment by Flamefire** [comment]
@branfosj Is this the same node as before? Suddenly **different** tests are failing :-(

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
n1503 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/1726a333baaa9c215c87c0842736dbb3 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
n1356 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/c1240667cbfd4de08c73568c7b98b655 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
n1498 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/eb79ee2ee75b8e421ec830778c668b27 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 35 out of 41 (1 easyconfigs in total)
i8018 - Linux Rocky Linux 8.7, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/Flamefire/21ed01a76c043fce98ba9f06af721d34 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
n1141 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/1184e9cb2f4c026961e65f9dd59b8917 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
i8026 - Linux Rocky Linux 8.7, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/Flamefire/5c4ccbbd67d10e2ce1687eacadb447ce for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2
CORE_CNT=16

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
n1162 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/3642d893a09f719ca27c036316df627c for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03b - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/b5485dd316e62a25c0ccb8855f0a1e57 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
EB_ARGS="--sanity-check-only"

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @Flamefire!

**Comment by branfosj** [comment]
Going in, thanks @Flamefire!

**Comment by branfosj** [comment]
Going in, thanks @Flamefire!
